### ISSUE TITLE
Contracts are accepted even when monetary restrictions are not met

### ISSUE THREAD

**Issue by 5thHorseman** [jrossignol/ContractConfigurator#568]
If you have a negative number in advanceFunds and the same number (but positive) in a Funds requirement (so the contract will not even be offered if you don't have the funds), you can set up a situation where you have the funds, get the contract offered, spend those funds, and then still accept the contract even though you are now not meeting its requirements.

On the forums you mentioned correcting this by not allowing you to accept a contract if doing so would run your funds into the negative.

**Comment by 5thHorseman** [comment]
You may already be planning this, but could I request this idea be expanded to actually check ALL restrictions before allowing someone to accept a contract? I've been giving the thought of having the player "spend" science instead of (or along with) money in my contract pack and if Contract Configurator only checks money, that'll put me in the same situation I'm in now, only with Science.

**Comment by jrossignol** [comment]
Yeah, you convinced me - on click the contract will get checked - so if it's no longer valid they won't even see it (it may change in the summary, on the left, but so be it).

**Comment by jrossignol** [comment]
Fixed for next release.

**Comment by 5thHorseman** [comment]
This does not seem to be working with my restrictions. I can keep accepting contracts that were correctly available upon entering Mission Control, even well after I've gotten down to $0 in my account.

Here is my output log from a run where I started in a new save, cheated $250k in, upgraded mission control, and accepted 7 contracts that were all valid when I entered mission control, but became invalid upon accepting the 4th one. I then exited Mission Control so the contracts would complete, went back in, and grabbed 7 more contracts each of which should have not been available anymore.
[output_log.txt](https://github.com/jrossignol/ContractConfigurator/files/455314/output_log.txt)

**Comment by jrossignol** [comment]
D'oh!  That's because I got distracted by the immediate requirement check and forgot to implement the negative advance/sufficient funds check.  I've added that in, mind double checking the dev dll and seeing if it works?

**Comment by 5thHorseman** [comment]
No it's still doing the exact same thing. It lets me accept contracts even when they'll take me to 0 funds and when I'm at 0 funds.

**Comment by jrossignol** [comment]
I had a sign reversed (I think).  Sorry, usually I'd be better at testing this stuff, just focused on 1.2 at the moment.

**Comment by 5thHorseman** [comment]
Totally understandable, regarding 1.2.

However sadly the new DLL still doesn't work. I noticed it DOES re-check the contracts upon clicking the section in the "All" tab, but not upon clicking each contract or upon clicking "Accept" on a contract you shouldn't be able to accept. Maybe the code is just in the wrong place, or not in every place it needs to be?

**Comment by jrossignol** [comment]
One last time, if you don't mind.  If that still doesn't work, I'll set up a test case and give it a more thorough run through some time during the week.

**Comment by 5thHorseman** [comment]
I don't mind at all, however sadly it's still not working. Here's another output_log for you, and also my current contract pack (not the release, just a zip of the folder from the ContractPacks folder). All you need do is cheat in a little money (I click 50k twice) so you can upgade Mission Control to 7 active contracts, so you have some room to accept multiple contracts and use up your money. Engines tend to be the most expensive :)
[output_log.txt](https://github.com/jrossignol/ContractConfigurator/files/456019/output_log.txt)
[PartsUnlimited.zip](https://github.com/jrossignol/ContractConfigurator/files/456021/PartsUnlimited.zip)
